src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -589,15 +586,47 @@ async def generate_chroot(request: SetupPyChrootRequest) -> SetupPyChroot:
                 f"`entry_point='{entry_point.module}:main'. See {url}."
             )
         entry_point_requests.append(ResolvePexEntryPointRequest(binary[PexEntryPointField]))
+
     binary_entry_points = await MultiGet(
         Get(ResolvedPexEntryPoint, ResolvePexEntryPointRequest, request)
         for request in entry_point_requests
     )
+    source_with_binaries = dict()
```

Comment:
Also, better to use a dict literal `{}` https://www.pantsbuild.org/docs/style-guide#use-collection-literals.